src/engine/symbolic_execution.rs

```diff
@@ -76,11 +131,20 @@ pub enum SymbolicExecutionError {
     #[error("failed to decode instruction at PC: {0:#010x}")]
     InvalidInstructionEncoding(u64, DecodingError),
 
-    #[error("failed to compute satisfyability for formula")]
-    SatUnknown(SolverError),
+    #[error("fatal error in SMT solver")]
+    Solver(SolverError),
+
+    #[error("found a bug")]
+    BugFound(Bug),
+
+    #[error("exit syscall called with code: {0}")]
+    ProgramExit(u64),
```

Comment:
nit: It seems this is only ever used with exit code zero. We could just call it `RegularProgramExit` and drop the argument. No strong opinion here, just and idea. Feel free to ignore.

Reply:
Hmm yeah I think we should talk about `exit_code != 0` anyway. Personally I wouldn't classify a program returning non zero as a bug. I think, I copied that definition from the original monster of Christoph. 
If we agree on that, I would like to remove the `Bug::ExitCodeGreaterZero` and use `ProgramExit` with all exit codes.

WDYT?

Reply:
In general I would be fine with this redefinition of a "bug", we just need to make sure that all test cases are adapted and use another canonical "bug" (like division by zero) in their assertions. The name `RegularProgramExit` would still be correct, but there is less need for the longer name, the argument we would need to keep of course. But lets discuss this more broadly first.

The gist: keeping `ProgramExit` as is in this PR is fine with me.

Reply:
Okay, let's chat about that.